<h1>We Rate Dogs: Report</h1>
<h4>By Luis Alarcon</h4>

<h2>Introduction</h2>

<p>The project belongs to the second term of the Udacity Data Analyst Nanodegree. This project starts with three main datasets:</p>

<ul>
    <li><b>Image Predictions</b>: Udacity provides this file.</li>
    <li><b>Twitter Archive</b>: Udacity provides this file.</li>
    <li><b>Twitter related to WeRateDogs</b>: This is obtained through a Twitter Development API.</li>
</ul>

<h3>Datasets: General Characteristics</h3>

<h4>Image Predictions</h4>
<p>The image predictions dataset has 2075 entries, with 12 columns with an assortment of types.</p>

<h4>Twitter Archive</h4>
<p>The Twitter Archive dataset has 2356 entries with 17 columns with an assortment of types.</p>

<h4>Twitter JSON</h4>
<p>The Twitter JSON dataset has 2341 entries with seven selected columns with an object and int64 types. Also, in order to gather the data for this dataset, we distinguished whether the twitter ID from the Twitter Archive dataset was found using the Twitter Development API; if not found then the twitter ID will be stored in a separate file.</p>

<h3>Quality Analysis</h3>

<h4>Null Values</h4>

</p>There are no null values in any of the datasets gathered.</p>

<h4>Data Problems</h4>

<p>After analysing the Twitter Archive dataset, some issues were found:</p>

<ul>
    <li><b>tweet_id:</b> The data type needs to be changed from int64 to object.</li>
    <li><b>in_reply_to_status_id:</b> erroneous data types.</li>
    <li><b>in_reply_to_user_id:</b> erroneous data types.</li>
    <li><b>name:</b> check if the name of the dog is the one mentioned in the tweet text. if it is not found remove it</li>
    <li><b>No retweets:</b> remove retweet data as well as columns associated with retweets.</li>
    <li><b>tweets with ratings with images.</b></li>
    <li><b>timestamp:</b> Change to datetime data type</li>
    <li><b>doggo, floofer, pupper, puppo:</b> change to one variable.</li>
    <li><b>rating:</b> Create a new column name rating which is the result of rating_numerator/rating_denomintor.</li>
</ul>

<p>All the errors found were amended.</p>

<p>Another dataset that presented issues after the initial analysis was the Image Prediction dataset. These were the issues found:</p>

<ul>
    <li><b>p1:</b> Some of the breeds predicted are not breeds. Therefore we will find whether the breeds predicted belong from a list made.</li>
    <li><b>p2:</b> Some of the breeds predicted are not breeds. Therefore we will find whether the breeds predicted belong from a list made.</li>
    <li><b>p3:</b> Some of the breeds predicted are not breeds. Therefore we will find whether the breeds predicted belong from a list made.</li> 
</ul>

<p>All issues found were amended/</p>

<p>Lastly, we fused the Twitter Archive and the Twitter JSON file, and we stored all modifications into new DataFrames and new files.</p>

<h3>Objectives</h3>

<p>The objective of this project is to understand the data provided to the students. The main focus of this report will be to analyse the primary dataset for values within the favourite count, retweet count and how that does interact with the rating provided, as well as the rating distribution.</p>

<p>Another aspect that we will only look at is the dog breed information in the image prediction file.</p>

<h2>Data Analysis</h2>

<p>The analysis started by finding the standard analysis of the Dataframe w_t_archive or working Twitter Archive file; the fusion between the Twitter Archive and Twitter JSON file.</p>


In [7]:
import pandas as pd

w_t_archive=pd.read_csv("./datasets/final_twitter-archive-enhanced.csv")
df_i_pre=pd.read_csv("./datasets/final_image-predictions.csv")

In [8]:
w_t_archive.describe()

tweet_id  in_reply_to_status_id  in_reply_to_user_id       rating  \
count  1.447000e+03           1.000000e+00         1.000000e+00  1447.000000   
mean   7.417931e+17           6.678065e+17         4.196984e+09     1.192581   
std    6.856868e+16                    NaN                  NaN     4.648617   
min    6.660293e+17           6.678065e+17         4.196984e+09     0.200000   
25%    6.782950e+17           6.678065e+17         4.196984e+09     1.000000   
50%    7.186133e+17           6.678065e+17         4.196984e+09     1.100000   
75%    7.961149e+17           6.678065e+17         4.196984e+09     1.200000   
max    8.924206e+17           6.678065e+17         4.196984e+09   177.600000   

       favorite_count  retweet_count  
count     1447.000000    1447.000000  
mean      9001.818936    2609.901175  
std      11882.749774    3948.182683  
min        105.000000      20.000000  
25%       2150.000000     638.500000  
50%       4495.000000    1433.000000  
75%      11536.000000    3146.000000  
max     123034.000000   60177.000000

<p>It shows that the mean rating is 1.93 and the mean favourite count is 9001.8, and the mean retweet count is 2609.90.</p>